core/deployment/src/main/java/io/quarkus/deployment/dev/QuarkusCompiler.java

```diff
@@ -164,6 +152,32 @@ public ClassLoaderCompiler(ClassLoader classLoader,
         }
     }
 
+    public void setupSourceCompilationContext(DevModeContext context, Set<File> classPathElements, DevModeContext.ModuleInfo i,
+            DevModeContext.CompilationUnit compilationUnit, String name) {
+        if (!compilationUnit.getSourcePaths().isEmpty()) {
+            if (compilationUnit.getSourcePaths() == null) {
```

Comment:
This condition would always be false, otherwise the line above would result in an NPE.